docs/src/replica_cluster.md

```diff
@@ -1,102 +1,382 @@
 
-- a private cloud spanning over multiple Kubernetes clusters in different data
-  centers
-- a public cloud spanning over multiple Kubernetes clusters in different
-  regions
-- a mix of the previous two (hybrid)
-- a public cloud spanning over multiple Kubernetes clusters in different
-  regions and on different Cloud Service Providers
+At this stage, the former primary has transitioned to a replica cluster,
+awaiting WAL data from the new global primary: `cluster-eu-central`.
 
-## Setting up a replica cluster
+To proceed with promoting the other cluster, you need to retrieve the
+`demotionToken` from `cluster-eu-south` using the following command:
 
-To set up a replica cluster from a source cluster, we need to create a cluster YAML
-file and define the following parts accordingly:
+```sh
+kubectl get cluster cluster-eu-south \
+  -o jsonpath='{.status.demotionToken}'
+```
 
-- define the `externalClusters` section in the replica cluster
-- define the bootstrap part for the replica cluster. We can either bootstrap via
-  streaming using the `pg_basebackup` section, or bootstrap from a volume snapshot
-  or an object store using the `recovery` section
-- define the continuous recovery part (`.spec.replica`) in the replica cluster. All
-  we need to do is to enable the replica mode through option `.spec.replica.enabled`
-  and set the `externalClusters` name in option `.spec.replica.source`
+!!! Note
+    The `demotionToken` obtained from `cluster-eu-south` will serve as the
+    `promotionToken` for `cluster-eu-central`.
 
-#### Example using pg_basebackup
+You can verify the role change using the `cnpg` plugin, checking the status of
+the cluster:
 
-This **first example** defines a replica cluster using streaming replication in
-both bootstrap and continuous recovery. The replica cluster connects to the
-source cluster using TLS authentication.
+```shell
+kubectl cnpg status cluster-eu-south
+```
+
+### Promoting a Replica to a Primary Cluster
+
+To promote a PostgreSQL replica cluster (e.g., `cluster-eu-central`) to a
+primary cluster and make the designated primary an actual primary instance,
+you need to perform the following steps simultaneously:
+
+1. Set the `.spec.replica.primary` to the name of the current replica cluster
+   to be promoted (e.g., `cluster-eu-central`).
+2. Set the `.spec.replica.promotionToken` with the value obtained from the
+   former primary cluster (refer to ["Demoting a Primary to a Replica Cluster"](replica_cluster.md#demoting-a-primary-to-a-replica-cluster)).
+
+The updated `replica` section in `cluster-eu-central`'s spec should look like
+this:
+
+```yaml
+replica:
+  primary: cluster-eu-central
+  promotionToken: <PROMOTION_TOKEN>
+  source: cluster-eu-south
+```
+
+!!! Warning
+    It is crucial to apply the changes to the `primary` and `promotionToken`
+    fields simultaneously. If the promotion token is omitted, a failover will be
+    triggered, necessitating a rebuild of the former primary.
```

Comment:
yikes. I hope we offer a plugin command to handle making this change safely

Reply:
yes, we need to create a `kubectl cnpg` subcommand for this